src/torrent.cpp

```diff
@@ -8283,7 +8277,7 @@ namespace libtorrent
 
 		int ret = 0;
 
-		std::int64_t const fin_time = finished_time();
+		std::int64_t const fin_time = duration_cast<seconds>(finished_time()).count();
```

Comment:
I cast it because the time ration except seconds. If both are the same duration I could just use count.

I would like to place a custom duration type to because at performance critical places like this sort criteria I want to skip the cast to same ration type.
During declaring a custom time I would use the opportunity to declare it with a size matching to its use-case. I'm not aware of the implications of stripping a 32bit type down by some bits or by raising a smaller 16bit type by some bits.
Something like 
`typedef std::chrono::duration<int:28, std::ratio<1, 1>> sec;`
which is enough for  -4 years to + 4 years
Or isn’t it worth it because of the byte fragmentation? And thats not a valid declaration.